lib/rules/react-prefer-private-members.js

```diff
@@ -85,9 +89,14 @@ function isReactStaticProperty({key: {name}}) {
     'contextTypes',
     'childContextTypes',
     'defaultProps',
+    'displayName',
   ].some((method) => method === name);
 }
 
+function isOtherAcceptedProperty({key: {name}}) {
```

Comment:
I would just check if this is a constructor, I think that also has a different node type